SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -142,17 +119,25 @@ void USpatialReceiver::OnAddComponent(Worker_AddComponentOp& Op)
 		GlobalStateManager->LinkAllExistingSingletonActors();
 		return;
 	case SpatialConstants::DEPLOYMENT_MAP_COMPONENT_ID:
- 		GlobalStateManager->ApplyDeploymentMapData(Op.data);
+		GlobalStateManager->ApplyDeploymentMapData(Op.data);
 		return;
 	case SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID:
- 		GlobalStateManager->ApplyStartupActorManagerData(Op.data);
+		GlobalStateManager->ApplyStartupActorManagerData(Op.data);
+		return;
+	}
+
+	if (ClassInfoManager->IsSublevelComponent(Op.data.component_id))
+	{
 		return;
-	default:
-		Data = MakeShared<improbable::DynamicComponent>(Op.data);
-		break;
 	}
 
-	PendingAddComponents.Emplace(Op.entity_id, Op.data.component_id, Data);
+	if (!bInCriticalSection)
+	{
+		UE_LOG(LogSpatialReceiver, Warning, TEXT("Received a dynamically added component, these are currently unsupported - component ID: %u entity ID: %lld"),
+			Op.data.component_id, Op.entity_id);
+		return;
```

Comment:
I believe this is wrong. If a client receives an add component for an already existing Actor, it should apply the data within this component.

Reply:
Yep, I agree. I'm going to add that today.